keepalive/keepalive.go

```diff
@@ -67,3 +67,12 @@ type ServerParameters struct {
 	// the connection is closed.
 	Timeout time.Duration
 }
+
+// EnforcementPolicy is used to set keepalive enforcement policy on the server-side.
+// Server will close connection with a client that violates this policy.
+type EnforcementPolicy struct {
```

Comment:
Should we have some notes in the ClientParameters to the effect of "make sure you set these parameters in coordination with the service owners, as incompatible settings can result in RPC failures"?

Especially since there's a default enforcement policy of 5m and no pings without streams, even if the service doesn't declare a policy.